### PR TITLE
Upgrade nextjs 12 -> 13

### PR DESCRIPTION
**Pull request by rstackhouse**
#44 

@gregv, this is to update the frontend from next 12 to 13. I tried to limit the scope of this change to doing what was strictly necessary to get all of the various pages to work in next 13.

There are quite a few potential improvements that I saw along the way. Most notably enabling more of the pages to be rendered server side.

I left TODOs in a number of places, but they would be extensions to existing functionality rather than losses thereof.

TODO:
- Set up MUI components for server side rendering. Currently this is blocking the SEO concern
- Use media queries rather than width comparisons for hiding labels in various places. Currently this is blocking SEO
- Improve SEO by relying more on server side rendering. Bring about info into page components

### CODE REVIEW

**Comment by rstackhouse** [comment]
@gregv, regarding server side rendering or not, the thing explicitly blocking this is hydration issues. Everywhere I had to use `dynamic`, the pre-rendered markup differed from the markup on the client.

File: src/components/OHackFeatures/OHackFeatures.js [hunk 1]
```diff
@@ -44,47 +47,51 @@ function a11yProps(index) {
 	};
 }
 
-function OHackFeatures() {
+const CAROUSEL_SELECTOR = '#ohack-feature-carousel';
+
+const OHackFeatures = () => {
+	// TODO: Move to the database. Have an admin form to edit these.
```
Comment (gregv) [inline src/components/OHackFeatures/OHackFeatures.js:53]: Love it

File: src/hooks/use-profile-api.js [hunk 2]
```diff
@@ -80,6 +84,7 @@ export default function useProfileApi(){
             if (options.authenticated) {
                 const token = await getAccessTokenSilently();
 
+                // TODO: auth is a cross-cutting concern. Add axios interceptor in _app.js get token out of localStorage and do this.
```
Comment (gregv) [inline src/hooks/use-profile-api.js:87]: This would definitely generalize how we're handling auth - good call out

File: src/pages/_app.js [hunk 3]
```diff
@@ -7,19 +9,28 @@ import theme from "../assets/theme";
...
+
+const Footer = dynamic(() => import('../components/Footer/Footer'), {
+  ssr: false,
+});
 import Head from "next/head";
 
-import * as ga from "../lib/ga";
-import { useRouter } from "next/router";
-import { useEffect } from "react";
+// import * as ga from "../lib/ga";
```
Comment (gregv) [inline src/pages/_app.js:24]: @rstackhouse - I think this removes Google Analytics from the website - I don't want to merge into main without a solve here

**Review by gregv — APPROVED** [review]
Lots of next/dynamic changes which lazily load things
There was a change to ga.js in `_app.js` so we'll want to make sure this is working before merging into main

> Next.js supports lazy loading external libraries with import() and React components with next/dynamic. Deferred loading helps improve the initial loading performance by decreasing the amount of JavaScript necessary to render the page. Components or libraries are only imported and included in the JavaScript bundle when they're used.

**Comment by gregv** [comment]
Getting build failure in Heroku
```
Failed to compile.
./src/components/Footer/Footer.js
```

**Comment by gregv** [comment]
This is now live on test.ohack.dev @rstackhouse !

**Pull request merged**